pkgs/development/python-modules/trainer/default.nix

```
@@ -30,6 +32,16 @@ buildPythonPackage {
     hash = "sha256-z6TOzWqE3NytkdG3nUzh9GpFVGQEXFyzSQ8gvdB4wiw=";
   };
 
+  patches = lib.optionals (pythonAtLeast "3.12") [
```

```suggestion
  patches = [
```
Patches should be applied as unconditionally as possible.
Otherwise, they may make maintenance unnecessarily hard.

---

Resolved in latest